commit c5e8bed1f0781b5296a5fb282615d19f2511ed91 (refs/changes/08/4008/1)
Author: Josh Wilsdon <josh@wilsdon.ca>
Date:   2018-05-25T11:13:07-07:00 (1 year, 4 months ago)
    
    TRITON-432 cn-agent should ignore non-log entries in stdio.log

diff --git a/lib/backends/smartos/lib/docker-stdio.js b/lib/backends/smartos/lib/docker-stdio.js
index 9a6289f..023e0cf 100644
--- a/lib/backends/smartos/lib/docker-stdio.js
+++ b/lib/backends/smartos/lib/docker-stdio.js
@@ -1304,7 +1304,9 @@ function runContainerLogsCommand(container, params, socket) {
     });
 
     lstream.on('readable', function runContainerLogsCommandOnReadable() {
+        var data;
         var line;
+        var rec;
 
         while ((line = lstream.read()) != null) {
             line = line.trim();
@@ -1318,20 +1320,21 @@ function runContainerLogsCommand(container, params, socket) {
                 break;
             }
 
-            var rec = JSON.parse(line);
-            var data;
-            if (params.Timestamps) {
-                data = rec.time + ' ' + rec.log;
-            } else {
-                data = rec.log;
-            }
+            rec = JSON.parse(line);
+            if (rec.log) { // only include entries that are stdio logs
+                if (params.Timestamps) {
+                    data = rec.time + ' ' + rec.log;
+                } else {
+                    data = rec.log;
+                }
 
-            writeData({
-                log: log,
-                streamType: rec.stream,
-                stream: socket,
-                chunk: data.toString()
-            });
+                writeData({
+                    log: log,
+                    streamType: rec.stream,
+                    stream: socket,
+                    chunk: data.toString()
+                });
+            }
         }
     });
 
